design/collision.layers.v2.md

```diff
@@ -0,0 +1,60 @@
+## One Line Concept:
+Change collision layer system to be more intuitive and safely support group masks and exclusive actors, UI, and triggers
+
+## Problem Description: 
+The collision system is used in a number of different cases, such as: Rigid Body Collision with Kinematic/Static objects or with other rigid bodies, Player Controller collision, Teleport ray collisions, UI ray collisions, proximity grab, Trigger Zones (for rigid bodies and for players.)
+
+The current collision layer implementation loosely wraps the Unity collision layers, and allowing 4 different layers (Default, NavMesh, Hologram, UI) to be assigned freely to all colliders regardless of usage.
+
+The design had to minimize number of collision layers (as that is a shared resource in Unity, and there are only 32 layers shared between all collision and all rendering), and that affected the design.
+
+The current implementation is problematic, because
+1. it's not intuitive what the different types do, or when they should be used.
+2. if using group masked/exclusive actors in certain cases, they can cause issues with rigid body simulation not being (largely) globally consistent across all connected clients
+
+I propose a new collision layer system that follow a number of requirements
+1. certain things (collisions between rigid body actors) depend on the authoritative peer client, and therefore exclusive actors and group masking cannot be allowed.
+2. anything that can happen on a specific client must be independent of the authoritative peer, to support exclusive actor and group masks - for example behaviors, personal UI, player-specific collisions and triggers
+3. each layer is configured for specific use cases and named accordingly.
+
+The system changes MRE collision layers to be more than just the equivalent of a Unity Layer, by adding additional restrictions and associated properties to each. Each layer has different requirements for
+1. Whether they can be exclusive/groupmasked, or they have to be global
+2. Which other layers it collides against
+3. Whether the collider must be triggers or non-triggers
+4. What it is generally used for
+
+I have come up with a list of 6 distinct collision layers to cover the common use cases. 
+
+ MRE Collision Layer Name|Used for|Can be exclusive/groupmasked|In Altspace uses Unity Layer|In Altspace Collides against Layers|IsTrigger in Unity
+ :--|:-:|:-:|:-:|:-:|--:
+ Physical|Rigid body, Player Collision, Blocks Cursor|no*|Default|Default, NavMesh, CharacterController, Interactables|no
```

Comment:
>CharacterController [](start = 84, length = 19)

Should physics interact with the player? Our player colliders are not very tight, it could be a weird experience.